lib/spack/spack/test/conftest.py

```diff
@@ -563,12 +563,10 @@ def _install(spec):
         pkg = spack.repo.get(s)
         pkg.do_install(fake=True, explicit=True)
 
-    # Transaction used to avoid repeated writes.
-    with mock_db.write_transaction():
-        _install('mpileaks ^mpich')
-        _install('mpileaks ^mpich2')
-        _install('mpileaks ^zmpi')
-        _install('externaltest')
+    _install('mpileaks ^mpich')
```

Comment:
As far as I reconstructed, this context manager was used since #2502.  Initially it was a way to speed-up the fixture by grabbing a write lock early instead of repeatedly take and release the lock for each installation (like Spack would do in its normal operations). This made sense since `database` or `mutable_database` were installing and uninstalling specs at least once per python file. Since some time though the installations are done just once per session, moved to a cache directory and copied wherever they are needed without the additional burden of installing and uninstalling them again. *All this is just to say that we will not lose anything in terms of test performance by removing the context manager.*

Instead, the *necessity* to remove it in this PR comes from the additional serialization I do of attributes related to the store. When we fork a process and we `_serialize` we create an equivalent, but different, store object in the forked process. If a write lock for the same DB is held by the parent process that will prevent the child to proceed.